deepchecks/vision/checks/train_test_validation/new_labels.py

```diff
@@ -85,10 +54,32 @@ def __init__(
 
         self.max_images_to_display_per_label = max_images_to_display_per_label
         self.max_new_labels_to_display = max_new_labels_to_display
+        self._display_images = defaultdict()
 
     def update(self, context: Context, batch: Batch, dataset_kind):
         """No additional caching required for this check."""
-        pass
+        if dataset_kind == DatasetKind.TRAIN:
+            pass
+        data = context.get_data_by_kind(dataset_kind)
+        images = data.batch_to_images(batch)
+        labels = data.batch_to_labels(batch)
+
+        for image, label in zip(images, labels):
+            if data.task_type == TaskType.CLASSIFICATION:
+                self._update_images_dict(label, label.item(), image)
+            elif data.task_type == TaskType.OBJECT_DETECTION and len(label) > 0:
+                for class_id in np.unique(label[:, 0].numpy()):
+                    bbox_of_label = label[(label[:, 0] == class_id).nonzero().squeeze(1)]
+                    self._update_images_dict(bbox_of_label, class_id, image)
+            elif len(label) > 0:
+                raise DeepchecksValueError(f'Unsupported task type {data.task_type.value} for NewLabels check')
+
+    def _update_images_dict(self, label, class_id, image):
```

Comment:
Should keep the indices for these images and return them as well, like in outliers.
Can be separate issue.